src/trix/elements/trix_editor_element.js

```diff
@@ -410,6 +412,7 @@ export default class TrixEditorElement extends HTMLElement {
     } else if (this.parentNode) {
       const toolbarId = `trix-toolbar-${this.trixId}`
       this.setAttribute("toolbar", toolbarId)
+      this.toggleAttribute("internal-toolbar", true)
```

Comment:
This line toggles the `[internal-toolbar]` attribute on the the `<trix-editor>` (since `this` references that element).

However, the `removeInternalToolbar()` function defined below checks `this.toolbarElement` (the `<trix-toolbar>` if the `<trix-editor>` element is using an "internal" toolbar).

Should this line set the attribute on the toolbar, or should the check be switched from `this.toolbarElement` to `this`?

Reply:
Furthermore, is it important that the attribute be set? Could this same kind of internal state tracking be achieved by a JavaScript object property that isn't represented in the DOM?

Reply:
I fixed and used an internal attribute to grab the internal toolbar. That's better indeed. Also added another test for the case of the external toolbar.